Portfolio -Jasmin Hedlund

JasminHed opened the pull request:
Netlify link: https://jasminh-portfolio.netlify.app/

Figma Link: https://www.figma.com/proto/SOYhLVq4NmWMK2nODwCj19/Figma?node-id=1045-1870&t=ivpFA7b8I2xj6r25-0&scaling=min-zoom&content-scaling=fixed&page-id=1045%3A3

Commit:
basic structure and code set up

Commit:
foldername

Commit:
updated footer

Commit:
added skiptolink

Commit:
tailwind

Commit:
basic styling card + project + journey + skill + tech

Commit:
added styling footer and hero + skiptolink

Commit:
styled mediaq in hero + footer and added in global styling main + mediaq

Commit:
created buttons + mobile first for journey+project

Commit:
updated path card

Commit:
created netlifytoml

Commit:
cardtest

Commit:
cardnetlify

Commit:
newcard

Commit:
updated mediq for journey and project, image, grid to fit, button styling, removed projects

Commit:
test

Commit:
testx

Commit:
testcard

Commit:
css on links

Commit:
styling for screens

Commit:
mainid

Commit:
updated p in hero

Commit:
images to webp

Commit:
removed redudant, updated sizes and text for better score at lighthouse

Commit:
Force add production build

Commit:
identation and h1 change

Commit:
nothing

Commit:
updated height + width in img

Commit:
updated arialabels

Commit:
removed auto height cardimage

Commit:
ogtags, lazy loading

Commit:
updated ogtag image

Commit:
optimized images

Commit:
removed old images, went through everything

Commit:
added cardwrapper + carousel+dots on desktop

Commit:
updated the dots for better accessability

Commit:
updated ammount of card on mobile and tablet

Commit:
checked accessability

Commit:
changed size on dots and hover on buttons + margin bottom+top on h2

Commit:
updated carousel like netflix

Commit:
added comments on desktop carousel

Commit:
updated color on arrow + sharper description

Commit:
updated readmefile

Commit:
added comments to advanced carousel for desktop to understand the code

Commit:
updated about me section and more after tip from mentor

Commit:
header, footer aria labels in app

JasminHed committed:
added happy thoughts project

JasminHed committed:
updated skills and intro

JasminHed committed:
added a blogpost, freelance

HIPPIEKICK reviewed (changes requested):
**A11y**
100% in Lighthouse ⭐

**Regarding the requirement: Your portfolio should follow the Figma design**
- On larger screens it doesn’t really look like the design layout-wise. Have a look at e.g. 1600px width and take action on aligning it all. Maybe set a max-width as you’ve done with the Skills section?
- There’s no box-shadow on the project cards in your design. If you want to have that - I’d suggest you add some more padding as it looks a bit cramped.
- Font-size of h2s are much smaller than in the design
- The gap between e.g. Skills section, projects/articles don’t adhere to the design

Some things to tweak to get to the finish line but you’re almost there 💪

**Codewise**
You have a good structure of your project and you’ve used components and props in a good way.

PS. Your dist folder should not be uploaded to GitHub.

**Changes requested**
Follow the Figma design. Almost there 💪

JasminHed committed:
shadowbox, better layout 1600px, new font h2 on desktop-not finished

JasminHed committed:
max-width main on desktop> +font-size on all H changed + boxshadow added project + more space on cards

JasminHed committed:
updated margin-bottom on all sections for space + figma design

JasminHed committed:
Remove dist from tracking

JasminHed commented:
Hello Matilda,
Many thanks for this feedback, it is really good to be able to "re-do" and look at it again and again. Great for learning! I have now tried to adjust according to your feedback :)

- Set a max-width in global styling for 1024-1600px
- Set box shadow on project cards + adjusted size and space to make it less crowded on larger screens
- Updated H2 (really all H) more in line with Figma but also accessibility on all screens
- Added margin bottom on all sections to create mor space according to Figma design
- Removed dist from github

Bianka2112 commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,53 @@
+    gap: 8px;
+  }
+`;
+
```
Very cool that you created global styles and can just inject the var for styling. If you wanted to save some lines of code here, use the method in your DesktopCarousel.jsx, line 73-ish. Where you can const a basic styled.a with all the sizing properties and then export const PrimaryButton = styled{basic} with your color choices or whatever applies only to that one. For example both buttons having a border (so they appear the exact same size) and then choosing what color each requires (same color as background for one, different for the other)

JasminHed replied:
Thanks for the details here, I am not sure what you mean but will take a look for sure! :)

Bianka2112 commented on src/components/Card.jsx:
```diff
@@ -0,0 +1,86 @@
+
+      {type === "project" && (
+        <>
+          <p>Languages: {data.languages.join(", ")}</p>
+          <p>Features: {data.features.join(", ")}</p>
```
Its cool how you structured your buttons based on section names. Seems very reusable this way =)
To further match your figma design, you can try wrapping a < strong > tag on Languages and Features to make them bold.

JasminHed replied:
Great idea, all implemented and done :)

Bianka2112 commented on src/components/Cardimage.jsx:
```diff
@@ -0,0 +1,12 @@
+
+const CardImage = styled.img`
+  width: 90%;
+  aspect-ratio: 4 / 3;
```
Didn't know we could write specific aspect-ratios like this! Nice 🤓

JasminHed replied:
Yeah its a great access!

Bianka2112 commented on src/components/DesktopCarousel.jsx:
This is amazing! I looked at your site first and was really impressed with the carousel feature 🙌 love that you included some comments breaking down whats happening here in code. Even though I dont 100% understand all of it- it looks great! Hope you are super proud 🥳

Bianka2112 commented on src/static-sections/Hero.jsx:
```diff
@@ -0,0 +1,112 @@
+    max-width: 1280px;
+    margin-bottom: 64px;
+  }
+`
```
I know I fought a lot with spacing in my own project. I think adding padding and margin is creating a bit too much space here.

JasminHed replied:
Thank you, all re done :)

Bianka2112 reviewed (commented):
Great looking portfolio Jasmin! Really love the carousel feature and how clean and accessible your site is. I would review the padding and margins in container sections to limit over spacing between sections. Also like the color choice as well! It feels very you and polished 😄 Hope I get a chance to hear how you setup the carousel and nice job on  all the articles you've written 🥳

p.s. I think its only to 1600px that we are graded, but you do seem to have your CardWrapper and DesktopCarousel Container both competing to show after 1601px if you wanted to check that. 🙌

JasminHed committed:
simplified carousel + new comments

JasminHed committed:
updated skills on 668px tablet

JasminHed commented:
Hello Bianka,
Many thanks for looking over my code! I appreciate you taking the time. I have done some changes based on your comments (also see those specific comments)
- Made feature and langugaes with a strong tag
-  Looked over spacing and margins in each section but especially hero/header
- Removed the mediaq that was set for 1601px, this was an emergancy solution that I now managed to solve in another way :) 
- As for the carousel, I did not have much choice as it was a part of my design :) but yes it was and is a great learning curve. I googled netflix carousels and then read up on some documentation. Actually i have already simplified it...a lot!..since you last looked. Sorry about that. But it started very detailed and complicated and I have then worked on the code in different sesion to make sure I understand the code snippets. It is a work in progress, happy to share what I have understood and done so far. 
Best
Jasmin

JasminHed committed:
removed mediaq 1601px, added strong tag on feature + language + updated hero margin and padding

HIPPIEKICK reviewed (approved):
Jasmin, I think you should put a max-width on the whole thing now that I think of it - do you agree? Here's an example of how it looks on my big screen with and without max-width (I just tried 1200px and I think it looks OK). And maybe let the footer (with the background) go all the way, but the rest cut off.

Do with that comment what you will 😄 The rest looks super so I'm approving!

Edit - I'm not a designer, so I don't know how to think about the carousel. Maybe they should go all the way to the right? 🤔 Just a thought.

![screencapture-jasminh-portfolio-netlify-app-2025-05-13-11_32_01](https://github.com/user-attachments/assets/9fe30682-a33c-44fb-9214-31856b8957bb)

![screencapture-jasminh-portfolio-netlify-app-2025-05-13-11_32_40](https://github.com/user-attachments/assets/8fad946f-802a-4486-a973-c487690cec4c)

JasminHed committed:
added max width on whole page 1200px, added happythoughts project + turned overflow visible on carousel for better responsiveness

JasminHed commented:
Hello Matilda,
OMG it looks great now with the width. Thank you for that! My brain could not see that possibility :) So:
- Made everything besides from footer max -width 1200px
- Made carousel overflow visible and that made it responsive and good looking. Yay!

JasminHed committed:
updated jurney and project data

JasminHed committed:
updated with freelance project UA

JasminHed committed:
added articles

JasminHed committed:
added fullstack project

JasminHed committed:
added fullstack project

JasminHed committed:
updated description

JasminHed committed:
saved updates

JasminHed committed:
updated projects text and features + added portfolio

JasminHed committed:
added a hamburger menu and menu w. home, projects + journey

JasminHed committed:
updated image on portofolio project

JasminHed committed:
new images

JasminHed committed:
updated image on project

JasminHed committed:
updated readmefile

JasminHed committed:
new article, + readme file -button missing

JasminHed committed:
landing page + new routes freelance projects

JasminHed committed:
updated journal

JasminHed committed:
new freelance

JasminHed committed:
updated filename

JasminHed committed:
removed routes, landingpage + freelancepart

JasminHed committed:
updated skills and intro

JasminHed committed:
added new project

JasminHed committed:
updated title for finalproject

JasminHed committed:
updated intro

JasminHed committed:
updated order of project and added freelance kodcentrum

JasminHed committed:
updated intro text

JasminHed committed:
updated repeating alt text

JasminHed committed:
fixed syntax error

JasminHed committed:
updated tools and upcoming

JasminHed committed:
updated toools

JasminHed committed:
darkmode, unique alttext to images, aria label to links, name on portfolio, list to all listitems

JasminHed committed:
removed skiptolink, added darkmode prefers depending on users screen, toogle works, moved darkmode to app, moved outside of main, unique linknames

JasminHed committed:
new links for ua and kodcentrum

JasminHed committed:
detailed styling of buttons

JasminHed committed:
new purple colors for darkmode, better alt text for wave

JasminHed committed:
new try

JasminHed committed:
new try w colors

JasminHed committed:
footercolor

JasminHed committed:
new

JasminHed committed:
new color + alt text

JasminHed committed:
updated description on ua and kodcentrum

JasminHed committed:
added new journal post on kodcentrum

JasminHed committed:
added kodcentrum project

JasminHed committed:
all projects now show

JasminHed committed:
added medium on AI and project

JasminHed committed:
new links

JasminHed committed:
updated intro

JasminHed committed:
updated intro

JasminHed committed:
updated intro

JasminHed committed:
updated intro
